v3/sqlite/sqlite.go

```diff
@@ -0,0 +1,256 @@
+func (c *conn) Prepare(query string) (driver.Stmt, error) {
+	return &stmt{conn: c, query: query}, nil
+}
+
+// Begin isn't supported.
+func (c *conn) Begin() (driver.Tx, error) {
+	return nil, errors.New("transactions are unsupported by this driver")
+}
+
+// connector implements driver.Connector.
+type connector struct {
+	conn *conn
+	name string
+}
+
+// Connect returns a connection to the database.
+func (d *connector) Connect(_ context.Context) (driver.Conn, error) {
+	if d.conn != nil {
+		return d.conn, nil
+	}
+	return d.Open(d.name)
+}
+
+// Driver returns the underlying Driver of the Connector.
+func (d *connector) Driver() driver.Driver {
+	return d
+}
+
+// Open returns a new connection to the database.
+func (d *connector) Open(name string) (driver.Conn, error) {
+	results := sqlite.ConnectionOpen(name)
+	if results.IsErr() {
+		return nil, toError(results.Err())
+	}
+	d.conn = &conn{
+		spinConn: *results.OK(),
+	}
+	return d.conn, nil
+}
+
+// Close closes the connection to the database.
+func (d *connector) Close() error {
+	if d.conn != nil {
+		d.conn.Close()
+	}
+	return nil
+}
+
+type rows struct {
+	columns []string
+	pos     int
+	len     int
+	rows    [][]any
+}
+
+var _ driver.Rows = (*rows)(nil)
+
+// Columns return column names.
+func (r *rows) Columns() []string {
+	return r.columns
+}
+
+// Close closes the rows iterator.
+func (r *rows) Close() error {
+	r.rows = nil
+	r.pos = 0
+	r.len = 0
+	return nil
+}
+
+// Next moves the cursor to the next row.
+func (r *rows) Next(dest []driver.Value) error {
+	if !r.HasNextResultSet() {
+		return io.EOF
+	}
+	for i := 0; i != len(r.columns); i++ {
+		dest[i] = driver.Value(r.rows[r.pos][i])
+	}
+	r.pos++
+	return nil
+}
+
+// HasNextResultSet is called at the end of the current result set and
+// reports whether there is another result set after the current one.
+func (r *rows) HasNextResultSet() bool {
+	return r.pos < r.len
+}
+
+// NextResultSet advances the driver to the next result set even
+// if there are remaining rows in the current result set.
+//
+// NextResultSet should return io.EOF when there are no more result sets.
+func (r *rows) NextResultSet() error {
+	if r.HasNextResultSet() {
+		r.pos++
+		return nil
+	}
+	return io.EOF // Per interface spec.
+}
+
+type stmt struct {
+	conn  *conn
+	query string
+}
+
+var _ driver.Stmt = (*stmt)(nil)
+var _ driver.ColumnConverter = (*stmt)(nil)
+
+// Close closes the statement.
+func (s *stmt) Close() error {
+	return nil
+}
+
+// NumInput returns the number of placeholder parameters.
+func (s *stmt) NumInput() int {
+	// Golang sql won't sanity check argument counts before Query.
+	return -1
+}
+
+func toRow(row []sqlite.Value) []any {
+	ret := make([]any, len(row))
+	for i, v := range row {
+		switch v.String() {
+		case "integer":
+			ret[i] = *v.Integer()
+		case "real":
+			ret[i] = *v.Real()
+		case "text":
+			ret[i] = *v.Text()
+		case "blob":
+			// TODO: check this
```

Comment:
Is this now TODONE?

Reply:
Good call, It's not. For some reason `blob` columns are being returned as `text`. I'm still tracing it.